### ISSUE TITLE
systemd v229 container networking is broken

### ISSUE THREAD

**Issue by mischief** [coreos/bugs#1205]
```
$ ssh coretest
CoreOS developer (1000.0.0+2016-04-04-1135)
Failed Units: 1
  usr.mount
core@coreos_production_qemu-1000-0-0- ~ $ docker run --rm -ti busybox /bin/sh -c "nslookup coreos.com"
Unable to find image 'busybox:latest' locally
latest: Pulling from library/busybox
385e281300cc: Pull complete 
a3ed95caeb02: Pull complete 
Digest: sha256:4a887a2326ec9e0fa90cce7b4764b0e627b5d6afcb81a3f73c85dc29cea00048
Status: Downloaded newer image for busybox:latest
Server:    10.0.2.3
Address 1: 10.0.2.3

nslookup: can't resolve 'coreos.com'
core@coreos_production_qemu-1000-0-0- ~ $ 
```

on a related note, upstream neutered IPForward=kernel, but in this case all forwarding options appear on. not sure what the problem is yet.

```
core@coreos_production_qemu-1000-0-0- ~ $ cat /proc/sys/net/ipv4/ip_forward
1
core@coreos_production_qemu-1000-0-0- ~ $ for f in /proc/sys/net/ipv4/conf/*/forwarding; do echo -n $f ": "; cat $f; done
/proc/sys/net/ipv4/conf/all/forwarding : 1
/proc/sys/net/ipv4/conf/default/forwarding : 1
/proc/sys/net/ipv4/conf/docker0/forwarding : 1
/proc/sys/net/ipv4/conf/eth0/forwarding : 1
/proc/sys/net/ipv4/conf/lo/forwarding : 1
```

**Comment by mischief** [comment]
it would appear that docker will bring up docker0 and assign an IP to it, and networkd will get a netlink message about it. when it does, it reads `/usr/lib64/systemd/network/50-docker.network` and due to there being an empty address assignment, it seems networkd is removing the bridge IP.

**Comment by mischief** [comment]
the following change to 50-docker.network appears to fix it, but this solution seems untenable.

```
[DHCP]
CriticalConnection=true
```

**Comment by mischief** [comment]
it appears systemd/systemd@5e5b137acc7094c9a1b8e7634b426dd445688bf0 changed this behavior.

**Comment by crawford** [comment]
Interesting. That would seem to imply that we can stop downing interfaces before we restart networkd in certain OEM configs.

**Comment by marineam** [comment]
My chat commentary:

00:53 < marineam> for documentation's sake, here is the commit that that breaks networking: https://github.com/systemd/systemd/commit/5e5b137acc7094c9a1b8e7634b426dd445688bf0
00:54 < marineam> this is a big fundimental change to how networkd behaves
00:54 < marineam> also, CriticalConnection is quite clearyly in the wrong section, it is applicale to all devices, not just DHCP configured ones.
00:56 < marineam> https://github.com/systemd/systemd/commit/e5d44b34cca39a335b8976328a14f47e2d7ef3b9 and 
                  https://github.com/systemd/systemd/commit/02e286219479ffa70ffeed5203d12e7fd831d9df further fixed up the behavior
00:57 < marineam> the ciritical connection option predates this, which explains why it is in the dhcp section: 
                  https://github.com/systemd/systemd/commit/eb27aeca247a4cf8816fffc4c0dbcab55ead3864

**Comment by marineam** [comment]
For a short term fix we should add a patch that just drops the call to `link_drop_foreign_config` but we need to figure out what to do long term.

**Comment by mischief** [comment]
fixed by coreos/systemd#45 as per @marineam's suggestion, but we might want to keep this issue open to discuss addressing this in a proper manner.

**Comment by crawford** [comment]
Let's discuss this in https://github.com/coreos/bugs/issues/1485.
